### ISSUE TITLE
Unable to join image/file to a post, because of CORS

### ISSUE THREAD

**Issue by Trim** [movim/movim#341]
Hello,

I'm testing Movim 0.11alpha1 on mov.adorsaz.ch and I've discovered an issue while I want to insert an image to my posts:

![Movim interface blocked on file upload](https://cloud.adorsaz.ch/index.php/s/w9n2c0Pklsl9Zwi/download)

On this screeshot, I've already choosen file to upload and I haven't any way to know that there was an issue.
I just see the "Cancel" button with the name of my file in the popup.

As you can see my setup is this one:

- Movim is intalled on subdomain mov.adorsaz.ch
- My XMPP server is installed on the domain adorsaz.ch
  - The HTTP File Server extension of my XMPP Server currently listen on domain adorsaz.ch and port 5280

According to the issue logged in developper console on the right, Movim has asked to my browser to secure connections made from this application to apply a Same Origin Policy for Cross-Origin requests.

I think that I'm not the only one with this issue, because, each user can uses his own XMPP HTTP File Server.

So, we need a solution to either:

- Add a HTTP Header Access-Control-Allow-Origin with the domain of the HTTP File Server used (as suggested by Firefox). This means the Movim web server have to receive some informations about the XMPP server before sending HTTP reply. I don't know if XMPP configuration details are known by the Movim web server.
- If not possible, block this feature when such siutation is detected
- If detection is not possible before, at least catch this error from the browser and warn the user he won't be able to use this feature. I think we can add a message to explain he have to upload the image on a public server and include it with public URL and MarkDown.

Thanks for your good work!

**Comment by Trim** [comment]
Good news: that error wasn't for Movim headers, but simply for the ejabberd http_upload module headers :)

Even better: this module already [allows to add custom headers](https://docs.ejabberd.im/admin/configuration/#modhttpupload). So I've modified my ejabberd configuration with this block

```yaml
   mod_http_upload:
     custom_headers:
        "Access-Control-Allow-Origin": "*"
        "Access-Control-Allow-Methods": "PUT, GET"
     docroot: "/srv/var/ejabberd/http_upload/"
     ....
```

And it works a lot better 👍

So now, Movim correctly sends my file, then it gets correctly the file URL and saves it in the link input, but finally says me there was an error on upload.

Although every thing is ok, no error log in console and my file is really available. I'll open another issue if I find more details about this strange behavior.

**Comment by Trim** [comment]
Sorry, I've closed the issue too fast, because when upload is not possible, the Movim UI should tell it without the need to open the javascript console.

**Comment by p0liper** [comment]
So? Do you fix this?

**Comment by p0liper** [comment]
Meanwhile, GHUC currently down :(
Your image, also, isn't open 

Offtop:
Request URL:https://camo.githubusercontent.com/****
Request Method:GET
Status Code:504 Gateway Timeout

**Comment by Trim** [comment]
GHUC seemes working now.

To summary:
I've fixed my configuration issue, but I let this bug report open, because the Movim interface dosen't advertise user there was an upload issue.

Currently, Movim keep the Upload popup opened with the file name selected and without any clue about an error.

I think that when an upload issue occurs, Movim should inform user and closes the Upload popup. Does it make sense ?

**Comment by edhelas** [comment]
https://enable-cors.org/
